p2panda-rs/src/wasm/tests/operation.rs

```diff
@@ -51,48 +49,78 @@ fn operation_fields() {
         )
         .unwrap();
 
+    // Make sure they have been added successfully
     assert_eq!(fields.get("name".to_string()).unwrap(), "Panda");
-    assert!(fields.get("is_panda".to_string()).unwrap());
-    assert_eq!(fields.get("age".to_string()).unwrap(), 5);
+    assert_eq!(fields.get("is_panda".to_string()).unwrap(), true);
     assert_eq!(fields.get("height_cm".to_string()).unwrap(), 167.8);
     assert_eq!(
         fields.get("favorite_cafe".to_string()).unwrap(),
         Hash::new_from_bytes(vec![1, 2, 3]).unwrap().as_str()
     );
+    assert_eq!(fields.len(), 4);
 
-    fields
-        .add(
-            "really_big_number".to_string(),
-            "int".to_string(),
-            JsValue::from_f64(31474836473453453434524523.0),
-        )
-        .unwrap();
-
-    assert!(fields.get("really_big_number".to_string()).is_err());
-
-    assert_eq!(fields.len(), 6);
-
-    fields.remove("really_big_number".to_string()).unwrap();
-
-    assert_eq!(fields.len(), 5);
-
+    // .. and remove them again successfully
     fields.remove("name".to_string()).unwrap();
     fields.remove("is_panda".to_string()).unwrap();
-    fields.remove("age".to_string()).unwrap();
     fields.remove("height_cm".to_string()).unwrap();
     fields.remove("favorite_cafe".to_string()).unwrap();
-
     assert_eq!(fields.len(), 0);
+}
 
+#[wasm_bindgen_test]
+fn inexistent_fields() {
+    let mut fields = OperationFields::new();
     let result = fields.remove("non_existant_key".to_string());
+    assert!(result.is_err());
+}
+
+#[wasm_bindgen_test]
+fn integer_fields() {
+    let mut fields = OperationFields::new();
+
+    // "int" fields get added as strings to allow large integers
+    fields
+        .add("age".to_string(), "int".to_string(), JsValue::from_str("5"))
+        .unwrap();
+
+    // "int" fields always get returned as BigInt instances
+    assert_eq!(
+        fields.get("age".to_string()).unwrap(),
+        JsValue::bigint_from_str("5")
+    );
+}
+
+#[wasm_bindgen_test]
+fn large_integers() {
+    let mut fields = OperationFields::new();
+
+    fields
+        .add(
+            "really_big_number".to_string(),
+            "int".to_string(),
+            JsValue::from_str("3147483647345534523"),
+        )
+        .unwrap();
 
+    assert_eq!(
+        fields.get("really_big_number".to_string()).unwrap(),
+        JsValue::bigint_from_str("3147483647345534523")
+    );
+
+    // This integer is too large and can't be represented as i64
+    let result = fields.add(
+        "really_big_number".to_string(),
+        "int".to_string(),
+        JsValue::from_str("932187321673219932187732188"),
+    );
     assert!(result.is_err());
 }
```

Comment:
Just so I'm clear, would this work if `JsValue::bigint_from_str("932187321673219932187732188")` was passed as the value instead?

Reply:
No, the `add` method only accepts strings for the "int" type. I've tried to use BigInts but I really can't see how to convert it into an i64 except of converting it to a string first and then to an integer which is weird, maybe you have an idea?